pysatSpaceWeather/instruments/sw_f107.py

```diff
@@ -148,69 +172,128 @@ def load(fnames, tag=None, inst_id=None):
     Called by pysat. Not intended for direct use by user.
 
     """
-    all_file_results = []
-    for fname in fnames:
-        if tag == '':
-            # F10.7 data is stored monthly an pysat needs daily data.  The
-            # daily date is attached to filename.  Parse off the last date,
-            # load month of data, downselect to desired day
-            date = dt.datetime.strptime(fname[-10:], '%Y-%m-%d')
-            data = pds.read_csv(fname[0:-11], index_col=0, parse_dates=True)
-            idx, = np.where((data.index >= date)
-                            & (data.index < date + pds.DateOffset(days=1)))
-            result = data.iloc[idx, :]
-        elif tag in ['all', 'daily', 'prelim', 'forecast', '45day']:
-            result = pds.read_csv(fname, index_col=0, parse_dates=True)
-        all_file_results.append(result)
-    # combine loaded data together
-    result = pds.concat(all_file_results, axis=0, sort=True)
-
+    # Get the desired file dates and file names from the daily indexed list
+    file_dates = list()
+    if tag == '':
+        unique_files = list()
+        for fname in fnames:
+            file_dates.append(dt.datetime.strptime(fname[-10:], '%Y-%m-%d'))
+            if fname[0:-11] not in unique_files:
```

Comment:
I'm curious, did you run into duplicate files or are you practicing general code safety?

Reply:
I forgot files were monthly (noticed in downloads). Ignore.